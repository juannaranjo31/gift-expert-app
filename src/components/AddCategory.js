import React, {useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');

    const handleSubmit = (e) => {
        e.preventDefault();
        
        if (inputValue.trim().length > 2) {
            setCategories( c => [inputValue, ...c]);
            setInputValue('');
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <input 
            type="text"
            value={inputValue}
            onChange={ (e) => setInputValue( e.target.value ) }
            placeholder='Ingrese una categoria para su busqueda de gifts :)'
            
            ></input>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
