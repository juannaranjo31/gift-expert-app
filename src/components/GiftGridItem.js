import React from 'react';
import PropTypes from 'prop-types';

export const GiftGridItem = ( {title, url} ) => {

    return (
        <div className='card animate__animated animate__fadeIn'>
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    )
}

GiftGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}