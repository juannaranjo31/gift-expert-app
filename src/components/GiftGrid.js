import React from 'react'
import PropTypes from 'prop-types';
import { useFetchGifts } from '../hooks/useFetchGifts';
import { GiftGridItem } from './GiftGridItem';

export const GiftGrid = ({ category }) => {

    const {data:images, loading} = useFetchGifts( category );

    return (
        <>
            <h3> {category} </h3>
            {loading && <p className='animate__animated animate__flash'>Loading...</p>}

            <div className='card-grid'>
                {
                    images.map((img) => {
                        return <GiftGridItem key={img.id} {...img} />
                    })
                }
            </div>

        </>
    )
}

GiftGrid.propTypes = {
    category: PropTypes.string.isRequired
};