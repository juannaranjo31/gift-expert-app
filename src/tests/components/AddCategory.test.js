import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";


describe('Pruebas sobre <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });

    test('Debe reenderizatse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });


    test('NO debe postear la informaciÃ³n con submit', () => {
        wrapper.find( 'form' ).simulate( 'submit', { preventDefault(){} } );

        expect( setCategories ).not.toHaveBeenCalled();
    });


    test('Debe llamar a setCategories y limpiar la caja de texto', () => {

        const value = 'One Punch';

        //Simular inputChange
        wrapper.find('input').simulate('change', { target: {value} });
        
        //Simular submit
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        //setCategories se debe haber llamado
        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledTimes(1);
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );
        expect( wrapper.find('input').prop('value') ).toBe( '' );
    });

});
