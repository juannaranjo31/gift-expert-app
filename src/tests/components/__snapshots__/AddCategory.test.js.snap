// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <AddCategory /> Debe reenderizatse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Ingrese una categoria para su busqueda de gifts :)"
    type="text"
    value=""
  />
</form>
`;
